radio/bluetooth/bluetooth.go

```diff
@@ -76,10 +79,66 @@ func Disconnect(client ble.Client) error {
 	return nil
 }
 
-func Scan(id string, timeout time.Duration) (map[string]bool, error) {
+func WriteChar(client ble.Client, char *ble.Characteristic, value []byte, noRsp bool) error {
+	err := make(chan error)
+	go func() {
+		err <- client.WriteCharacteristic(char, value, noRsp)
+	}()
+
+	for {
```

Comment:
From what I can tell here it looks like you are busy-waiting on a asynchronous task. This in general is a big anti-pattern and you should consider using other constructs to achieve the same results. Busy-waiting means that (in general) the kernel does not know that you are waiting for something and hence cannot take your process out of the scheduling queue. This means that rather than using 0% CPU, it will be a lot more, most likely an entire core.

I don't know Go well enough to suggest other constructs, but a quick google shows up https://golang.org/pkg/sync/#WaitGroup which would tell the CPU that you are blocking until the goroutine is finished. This means that the kernel will not schedule the process until the goroutine has finished.